src/common/actions/create-snap.js

```diff
@@ -54,10 +53,11 @@ function getSnapName(owner, name) {
     }));
 }
 
-export function createSnap(repositoryUrl, location) { // location for tests
+export function createSnap(repository) { // location for tests
```

Comment:
I discovered that all code calling createSnap already had a parsed repository object. Instead of parsing it twice I decided to just pass the repository object.